src/stk/image/volume.cpp

```diff
@@ -121,6 +178,8 @@ Volume Volume::as_type(Type type) const
 {
     ASSERT(valid());
     ASSERT(type != Type_Unknown);
+    // Non-contiguous not supported for now, this function is gonna be refactored anyway
+    ASSERT(is_contiguous());
```

Comment:
Couldn't be bothered to add the strides to this function since it's a mess and I'll probably rewrite it anyways.